polkadot/transaction-pool/src/lib.rs

```diff
@@ -53,37 +59,65 @@ pub use error::{Error, ErrorKind, Result};
 pub type CheckedExtrinsic = <UncheckedExtrinsic as Checkable>::Checked;
 
 /// A verified transaction which should be includable and non-inherent.
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub struct VerifiedTransaction {
-	inner: CheckedExtrinsic,
+	original: UncheckedExtrinsic,
+	inner: Mutex<Option<CheckedExtrinsic>>,		// `Some` only if the Address is an AccountId
 	hash: Hash,
 	encoded_size: usize,
 }
 
+impl Clone for VerifiedTransaction {
+	fn clone(&self) -> Self {
+		VerifiedTransaction {
+			original: self.original.clone(),
+			inner: Mutex::new(self.inner.lock().clone()),
+			hash: self.hash.clone(),
+			encoded_size: self.encoded_size.clone(),
+		}
+	}
+}
+
 impl VerifiedTransaction {
 	/// Attempt to verify a transaction.
-	fn create(xt: UncheckedExtrinsic) -> Result<Self> {
-		if !xt.is_signed() {
-			bail!(ErrorKind::IsInherent(xt))
+	fn create(original: UncheckedExtrinsic) -> Result<Self> {
+		if !original.is_signed() {
+			bail!(ErrorKind::IsInherent(original))
 		}
+		let (encoded_size, hash) = original.using_encoded(|e| (e.len(), BlakeTwo256::hash(e)));
+		let lookup = |a| match a {
+			RawAddress::Id(i) => Ok(i),
+			_ => Err("chain state not available"),
+		};
+		let inner = Mutex::new(match original.clone().check(lookup) {
+			Ok(xt) => Some(xt),
+			Err(e) if e == "chain state not available" => None,
```

Comment:
should extract out to constant (within the scope of this function) to avoid accidental differences